moondeck/scenario/run_live_scenario.py

```
@@ -460,6 +460,23 @@ def run_scenario(client: Client, scenario_path: Path, settle_s: float = 1.5,
+                        for key, value in (step.get("props") or {}).items():
+                            try:
+                                client.post("/api/control",
+                                            {"module": step.get("id", ""), "control": key,
+                                             "value": value})
+                            except urllib.error.HTTPError as pe:
+                                # A prop the target does not offer is the same not-available-here
+                                # case an optional set_control is, and fails the step otherwise.
+                                if not step.get("optional"):
+                                    raise
+                                print(f"  SET   {step.get('id','?')}.{key}: skipped "
+                                      f"(optional, not offered on {target}: {pe.code})")
```

Ruff (0.16.3)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)